src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -3041,6 +3043,33 @@ class CgroupUtils(object):
             # Note: sometimes normal (e.g. interactive jobs)
             pass
 
+    def set_diag_messages(self, resc_used, msg, concate):
+        """
+        Sets the 'diag_messages' resource to msg. A json format of msg
+        is required in order to merge msgs from sis moms and primary mom.
+        Settting concate=true concates multiple string values of the
+        same dict key into one.
+        """
+        try:
+            json_msg = json.loads(msg)
+        except Exception:
+            json_msg = {}
+        if 'diag_messages' not in resc_used.keys():
+            resc_used['diag_messages'] = json.dumps(json_msg)
+        elif concate:
+            try:
+                json_resc = json.loads(resc_used['diag_messages'])
+            except Exception:
+                json_resc = {}
+            for k in json_msg.keys():
+                if k in json_resc.keys() and not json_msg[k] in json_resc[k]:
+                    json_resc[k] += ', %s' % json_msg[k]
```

Comment:
I'm not sure about OpenPBS, but at least with commercial, we no longer support python releases older than 3.6.  A new type of string formatting was introduced in 3.6 called fstrings.  They are much clearer and easier to use than the old string formatting.

f"foo: {foo} bar: {bar} baz: {baz[5]}"
Anything between the {} is python code to be evaluated.  Note that when using a dictionary, you need to use alternate quotes f"{foo['bar']}"

Reply:
You are right. It is nicer. Moreover, we already use it in PTL. Changed.